Create an initial version of the script retention dashboard.

ashercodeorg opened the pull request:
![image](https://cloud.githubusercontent.com/assets/14206260/13410678/a7f6240a-defe-11e5-89d7-df7f16bcb9fd.png)

TODO(other PRs):
- Improve user-friendliness by converting script IDs to script names when displaying.
- Provide a Download to CSV option.
- Surface Properties.get(:retention_stats)['stage_level_counts'] in a stage-retention dashboard.

Hamms commented on dashboard/app/controllers/admin_reports_controller.rb:
```diff
@@ -259,6 +259,24 @@ def hoc_signups
+      # Get the cached retention_stats from the DB, trimming those stats to only the requested
+      # scripts.
+      @retention_stats = Properties.get(:retention_stats).each_pair do |key, key_data|
+        key_data.delete_if{|script_id, script_data| !@scripts.include? script_id.to_i}
```
Is there a reason you're not just using a `select`?

ashercodeorg replied:
Note that the select would be a ruby select, not a DB select.

Is
@retention_stats = {}
Properties.get(:retention_stats).each_pair do |key, key_data|
  @retention_stats[key] = key_data.select!{|script_id, script_data| @scripts.include? script_id.to_i}
end

preferred? Happy to change it to this if it is.

Hamms replied:
`key_data.select`, not `key_data.select!`, but otherwise yeah!

ashercodeorg replied:
Done.

Hamms commented on dashboard/app/controllers/admin_reports_controller.rb:
```diff
@@ -276,6 +294,26 @@ def get_percentage_positive(ratings_to_process)
+  # a slice across the hash arrays.
+  def build_row_arrays(count_stats)
+    # Determine the number of final number of rows, being the maximum array size in the hash.
+    array_length = count_stats.max_by{|k,v| v.length}[1].size
```
is there a reason you're using both `length` and `size`?

ashercodeorg replied:
No good reason (changed length to size).

Hamms commented:
Could I get some screenshots of what this looks like?

ashercodeorg commented:
Added a screenshot. Data isn't very interesting because my localhost doesn't have much data. Note that I also added edge case handling (fixing tests and alerting users to possible user error).

Hamms commented:
LGTM!
